src/main/java/com/iota/iri/hash/PearlDiver.java

```diff
@@ -178,28 +161,16 @@ public synchronized boolean search(final int[] transactionTrits, final int minWe
                                     (midCurlStateCopyLow[i] & outMask) == 0 ? 1
                                         : (midCurlStateCopyHigh[i] & outMask) == 0 ? -1 : 0;
                             }
-                            syncObj.notifyAll();
                         }
                     }
-                    break;
                 }
             }));
             workers[threadIndex] = worker;
+            worker.setName(this + ":worker-" + threadIndex);
+            worker.setDaemon(true);
             worker.start();
         }
```

Comment:
At least now, in an IDE or code analysis tool, you can easily identify the PearlDiver instance and worker numbers.
Also DAEMON status means that if the Java machine needs to exit, that it will kill these threads instead of waiting.

Reply:
:+1: